server/src/main/java/org/opensearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -493,7 +504,15 @@ public InternalAggregation reduce(List<InternalAggregation> aggregations, Reduce
         } else {
             docCountError = aggregations.size() == 1 ? 0 : sumDocCountError;
         }
-        return create(name, Arrays.asList(list), reduceContext.isFinalReduce() ? order : thisReduceOrder, docCountError, otherDocCount);
+
+        // Shards must return buckets sorted by key, so we apply the sort here
+        List<B> resultList;
+        if (reduceContext.isSliceLevel()) {
+            resultList = Arrays.stream(list).sorted(thisReduceOrder.comparator()).collect(Collectors.toList());
```

Comment:
We can use `Arrays.sort(list, thisReduceOrder.comparator());` to avoid copy of the elements.